### PR TITLE
chore: Move comments event handler to use proper event dispatcher

### PR DESCRIPTION
**Pull request by juliushaertl**
Similar to https://github.com/nextcloud/server/pull/45951 this moves away from the not so lazy registration of comment event handlers.

- [x] Requires https://github.com/nextcloud/server/pull/45951/files#diff-862a3988c96ee1f18bc17076a3152274d876dec178266a6b2d237b9ae7f03cd0R1534

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Tests (unit, integration, api and/or acceptance) are included
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by luka-nextcloud — DISMISSED** [review]
The failed cypress tests don't seem to be related to the code change.
The failed integration test is related to the code change. Seems like comment event was not triggered.

File: tests/unit/Activity/CommentEventHandlerTest.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function setUp(): void {
 		$this->notificationHelper = $this->createMock(NotificationHelper::class);
 		$this->cardMapper = $this->createMock(CardMapper::class);
 		$this->changeHelper = $this->createMock(ChangeHelper::class);
-		$this->commentEventHandler = new CommentEventHandler(
```
Comment (blizzz) [inline tests/unit/Activity/CommentEventHandlerTest.php:52]: needs a use statement

**Review by blizzz — CHANGES_REQUESTED** [review]

File: tests/unit/Listeners/CommentEventListenerTest.php [hunk 2]
```diff
@@ -21,8 +21,9 @@
  *
  */
 
-namespace OCA\Deck\Activity;
+namespace OCA\Deck\Listeners;
```
Comment (blizzz) [inline tests/unit/Listeners/CommentEventListenerTest.php:24]: :grimacing: okay i guess you can do it that way

**Review by blizzz — APPROVED** [review]

File: tests/unit/Listeners/CommentEventListenerTest.php (hunk 2 above)
Comment (juliushaertl) [reply tests/unit/Listeners/CommentEventListenerTest.php:24]: I also addressed the review comment from Luka with it as the namespace of the actual class was changed, it made sense to adapt the text namespace as well ;)

**Pull request merged**